src/main/java/com/plugin/frontend/actions/FormatFileAction.java

```diff
@@ -1,29 +1,37 @@
 package com.plugin.frontend.actions;
 
+import com.google.googlejavaformat.java.FormatterException;
 import com.intellij.openapi.actionSystem.AnAction;
 import com.intellij.openapi.actionSystem.AnActionEvent;
 import com.intellij.openapi.actionSystem.LangDataKeys;
+import com.intellij.openapi.editor.Document;
 import com.intellij.openapi.fileEditor.FileDocumentManager;
 import com.intellij.openapi.vfs.VirtualFile;
 import com.plugin.engine.JavaFormatter;
 import org.jetbrains.annotations.NotNull;
 
 import java.io.File;
-import java.util.Objects;
+import java.io.IOException;
 
 public class FormatFileAction extends AnAction {
     @Override
     public void actionPerformed(@NotNull AnActionEvent e) {
-        JavaFormatter test = new JavaFormatter();
-        VirtualFile file = FileDocumentManager.getInstance().getFile(Objects.requireNonNull(e.getData(LangDataKeys.EDITOR)).getDocument());
+        JavaFormatter javaFormatter = new JavaFormatter();
+        VirtualFile file = null;
+        try {
```

Comment:
Use the condition. This is better)

Reply:
I can’t use it here because .getDocument() cannot return null, it throws only NullPointerException